lib/galaxy/webapps/galaxy/controllers/history.py

```diff
@@ -578,7 +578,7 @@ def view(self, trans, id=None, show_deleted=False, show_hidden=False, use_panels
             else:
                 error_msg = ('An error occurred getting the history data from the server. '
                              + 'Please contact a Galaxy administrator if the problem persists.')
-            return trans.show_error_message(error_msg, use_panels=use_panels)
+            return {"error_msg": error_msg}
```

Comment:
I would do here something similar to
https://github.com/galaxyproject/galaxy/blob/7f0804d7206268baa891bb58341bfa8726664664/lib/galaxy/webapps/galaxy/controllers/history.py#L885-L887

That way you can get a proper status code in the response.
Also, in this particular case, since the problem is the history ID being invalid, I think is safe to return a little more specific error message saying so (just expect `MalformedId` exception), that may help reduce the "contact Galaxy administrator" part and probably admins will be happier :laughing:

Reply:
And you've actually API-fied the response, so you should be able to switch `expose` out to `expose_api` and drop the exception handling completely. The MalfomedId exception already has the details and the API decorator will turn that into a proper message.

Reply:
Yep, that's way better!

Reply:
Done, thanks @mvdbeek @davelopez!

Reply:
![image](https://user-images.githubusercontent.com/15801412/143586973-f75b69dc-8263-47a2-bfb3-027e170c367d.png)